javascript/javascript1/week4/voice_assistant.js

```diff
@@ -136,22 +156,58 @@ function calculator(a, b, operation) {
 }
 
 function calculateResult(input) {
-  const regex = /what is (\d+)\s*([+\-*\/])\s*(\d+)\?/i;
-  const match = input.match(regex);
+  const match = input.match(patterns.mathPattern);
   if (match && match.length === 4) {
-    const num1 = parseInt(match[1]);
+    const num1 = parseInt(match[1]); //extracts numbers and operation from a string
     const operation = match[2];
     const num2 = parseInt(match[3]);
     return calculator(num1, num2, operation);
   } else {
     return "Invalid input.";
   }
 }
-//console.log(greetings("Hello, my name is Tanya"));
-console.log(addToDo("Add dishes to my to do list."));
+
+function setTimer(input) {
+  const match = input.match(patterns.timerPattern);
+  if (match && match.length === 2) {
+    const minutes = parseInt(match[1]);
+    setTimeout(function () {
+      console.log("Time is up!");
+    }, minutes * 60 * 1000);
+    return `Timer is set for ${minutes} min.`;
+  } else {
+    return "Invalid input for setting a timer.";
+  }
+}
+function smartWear(input) {
+  const match = input.match(patterns.tempPattern);
+  if (match && match.length === 2) {
+    const temp = match[1];
+    const clothes = [
+      ["shorts and a t-shirt", [18, 48]],
+      ["jeans and a sweater", [12, 18]],
+      ["light coat", [5, 12]],
+      ["winter coat and a winter hat", [-5, 5]],
+    ];
+    for (let i = 0; i < clothes.length; i++) {
+      const tempRange = clothes[i][1];
+      if (temp >= tempRange[0] && temp < tempRange[1]) {
+        return clothes[i][0];
+      }
+    }
+    return `Stay home! It's freezing!`;
+  }
+}
```

Comment:
Nice refactor! Though it seems it crept into the voice_assistant.js commit. It looks like you're having fun with the math and logic which is great to see!